tests/test_encoding/test_similarity_encoder.py

```diff
@@ -213,3 +213,162 @@ def test_get_feature_names_out_na(df_enc_big_na):
     }
     assert tr.get_feature_names_out(input_features=None) == out
     assert tr.get_feature_names_out(input_features=input_features) == out
+
+
+def test_keywords_bad_type():
+    with pytest.raises(ValueError):
+        StringSimilarityEncoder(keywords="hola")
+
+
+def test_keywords_bad_items():
+    with pytest.raises(ValueError):
+        StringSimilarityEncoder(keywords={"column": "hola"})
+
+
+def test_keywords_dont_match(df_enc_big):
```

Comment:
Could we either add columns that are in the data to the dictionary, or even better, test a second dictionary with a combination of columns that are in the data, and a column that is not?

Also, here we are testing that the variables/columns don't match, correct? can we adjust the name of the test?